test/Table_Tests/src/Common_Table_Operations/Derived_Columns_Spec.enso

```diff
@@ -197,3 +197,29 @@ spec setup =
             t.set (Column_Operation.Add (Column_Ref.Name "X") (Column_Ref.Name "Z")) . should_fail_with No_Such_Column
             t.set (Column_Operation.Not (Column_Ref.Name "zzz")) . should_fail_with No_Such_Column
             t.set (Column_Operation.Not (Column_Ref.Index 42)) . should_fail_with Index_Out_Of_Bounds
+
+    Test.group "Unique derived column names" <|
+        Test.specify "Should disambiguate two derived columns that would otherwise have had the same name" <|
+            t = table_builder [["X", [1, 2, 3]]]
+            column_op = Column_Operation.Power 2 (Column_Ref.Name "X")
+            t2 = t.set column_op . set column_op
+            t2.column_names . should_equal ["X", "[2] ^ [X]", "[2] ^ [X] 1"]
+            t2.at "X" . to_vector . should_equal [1, 2, 3]
+            t2.at "[2] ^ [X]" . to_vector . should_equal [2, 4, 8]
+            t2.at "[2] ^ [X] 1" . to_vector . should_equal [2, 4, 8]
+
+        Test.specify "Should disambiguate two derived columns that would otherwise have had the same name, within the same expression" <|
+            t = table_builder [["X", [1, 2, 3]]]
+            expression = "2 + (2 * 2) + (2 ^ [X])"
+            t2 = t.set expression
+            t2.column_names . should_equal ["X", expression]
+            t2.at "X" . to_vector . should_equal [1, 2, 3]
+            t2.at expression . to_vector . should_equal [8, 10, 14]
+
+        Test.specify "Should use .pretty to distinguish string constants from regular column names" <|
+            t = table_builder [["X", ["a", "b", "c"]]]
+            expression = '"foo" + [X] + "bar"'
+            t2 = t.set expression
+            t2.column_names . should_equal ["X", expression]
+            t2.at "X" . to_vector . should_equal ["a", "b", "c"]
+            t2.at expression . to_vector . should_equal ["fooabar", "foobbar", "foocbar"]
```

Comment:
Ah, I forgot that with expressions the name is just the expression string. So this does not really test the `.pretty` logic...

Can we add one more test case? (Sorry for asking again and again)
```
t = table_builder [["X", ["a", "b", "c"]]]
t2 = t.set (Column_Operation.Add "prefix" (Column_Reference.Name "X"))
t3 = t2.set (Column_Operation.Add "prefix" "X")

t3.column_names . should_equal ["X", "['prefix'] + [X]", "['prefix'] + 'X'"]
t3.at "['prefix'] + [X]" . to_vector . should_equal ["prefixA", "prefixB", "prefixC"]
t3.at "['prefix'] + 'X'" . to_vector . should_equal ["prefixX", "prefixX", "prefixX"]
```

Reply:
Done